src/software/SfM/main_SfMInit_ImageListing.cpp

```diff
@@ -54,13 +59,71 @@ bool checkIntrinsicStringValidity(const std::string & Kmatrix, double & focal, d
       std::cerr << "\n Used an invalid not a number character" << std::endl;
       return false;
     }
-    if (i==0) focal = readvalue;
-    if (i==2) ppx = readvalue;
-    if (i==5) ppy = readvalue;
+    size_t component = i % 9;
+    if (component==0) focals.push_back(readvalue);
+    if (component==2) ppxs.push_back(readvalue);
+    if (component==5) ppys.push_back(readvalue);
   }
   return true;
 }
 
+bool checkImageDirsStringValidity(const std::string& sImageDirs, std::vector<std::string> &ImageDirs, std::string sCombinationDirectory) {
```

Comment:
change signature to std::vector<std::string> checkImageDirsStringValidity(const std::string &sImageDirs, const std::string &sCombinationDirectory)